pkg/api/handlers/libpod/containers.go

```diff
@@ -325,3 +326,84 @@ func ListContainerBatch(rt *libpod.Runtime, ctr *libpod.Container, opts shared.P
 	}
 	return ps, nil
 }
+
+func Checkpoint(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Keep           bool   `schema:"keep"`
+		LeaveRunning   bool   `schema:"leaveRunning"`
+		TCPEstablished bool   `schema:"tcpEstablished"`
+		Export         string `schema:"export"`
```

Comment:
We can't just pass this over - we'll be creating the export tar on the server. Expected behavior would be the client, I imagine. If export is requested we probably need to save the tar to a temporary file and stream it back to the client.